src/main/java/org/folio/rest/impl/PermissionUtils.java

```diff
@@ -42,5 +62,50 @@ public static boolean equals(OkapiPermission okapiPerm, Permission perm) {
         && Objects.equals(okapiPerm.getDescription(), perm.getDescription())
         && Objects.equals(okapiPerm.getDisplayName(), perm.getDisplayName())
         && Objects.equals(okapiPerm.getSubPermissions(), perm.getSubPermissions());
-  }  
+  }
+
+  /**
+   * Purge deprecated permissions for a given tenant.
+   * 
+   * @param pgClient {@link PostgresClient}
+   * @param tenantId tenant id
+   * @return {@link Future} with {@link PermissionNameListObject}
+   */
+  public static Future<PermissionNameListObject> purgeDeprecatedPermissions(PostgresClient pgClient, String tenantId) {
+    Promise<PermissionNameListObject> promise = Promise.promise();
+    pgClient.startTx(tx -> {
+      try {
+        pgClient.select(tx, String.format(SELECT_DEPRECATED_PERMS, tenantId), res -> {
+          if (res.failed()) {
+            pgClient.rollbackTx(tx, done -> promise.fail(res.cause()));
+            return;
+          }
+          PermissionNameListObject permNames = new PermissionNameListObject();
+          permNames.setTotalRecords(0);
+          @SuppressWarnings("rawtypes")
+          List<Future> futures = new ArrayList<Future>();
+          res.result()
+            .forEach(row -> {
+              String id = row.getString("id");
+              String name = row.getString("name");
+              futures.add(Future.<RowSet<Row>>future(p -> pgClient.execute(tx, String.format(PURGE_DEPRECATED_SUB_PERMS, tenantId, id, name, id, name), p)));
+              futures.add(Future.<RowSet<Row>>future(p -> pgClient.execute(tx, String.format(PURGE_DEPRECATED_PERMS_USERS, tenantId, id, name, id, name), p)));
+              futures.add(Future.<RowSet<Row>>future(p -> pgClient.execute(tx, String.format(PURGE_DEPRECATED_PERMS, tenantId), p)));
+              permNames.getPermissionNames().add(name);
+              permNames.setTotalRecords(permNames.getTotalRecords() + 1);
+            });
+          CompositeFuture.all(futures)
```

Comment:
Can we use https://github.com/folio-org/okapi/blob/master/okapi-common/src/main/java/org/folio/okapi/common/GenericCompositeFuture.java here yet?  I think the issues with dependency versions have been resolved w/ the upgrade to RMB 32